caffe2/core/predictor.cc

```diff
@@ -48,9 +48,11 @@ void Predictor::run(const TensorVector& inputs, TensorVector* outputs) {
 
   CAFFE_ENFORCE(ws_.RunNet(run_net_.name()));
 
-  outputs->resize(run_net_.external_output_size());
-  for (auto i = 0; i < outputs->size(); ++i) {
-    (*outputs)[i] = extractOutputTensor(&ws_, run_net_.external_output(i));
+  if (outputs != nullptr) {
```

Comment:
Oh I see. Thank you for such detailed explanation. I agree that to keep it simple, users shall not need to access the workspace, then outputs are necessary to be specified.

I will close this PR.